### PR TITLE
216 refactor colors

### PR DESCRIPTION
**Pull request by misteu**
### What does this do?
Refactors definition of colors into `Assets` file so that there is no additional setup needed for light and dark mode colors. The variants are now baked into the Assets file and loaded dynamically.

Also the Scribe key icon is placed into that file for each variant (light iPhone/iPad, dark iPhone/iPad), removing further logic.

### How does it work?
All colors can be accessed via the new `ScribeColors` enum. Each case`s `rawValue` refers to a color that can be accessed via `UIColor(named:....)`. To make it a little easier, there are two more convenient ways for initializing colors:

Both of the following statements will assign a `UIColor`.
1. Call new convenience initializer of `UIColor`, e.g.: `let color = UIColor(.keyPressed)`
2. Call via the enum itself: `let color = ScribeColor.keyPressed.color`

When refactoring `ColorVariables.swift` I only used the first variant. If you think the other one is not needed we could also completely remove it.

I decided against adding the colors as `static let` properties of  `UIColor` to not mix up the custom colors with the system ones. If you think they _should_ be part of that namespace, it would be a quick change.

There are small extensions added to quickly get light or darkmode variants of any color as these were explicitly accessed before as well (Not sure if intended but to have the same behavior I kept it like that):

You can get any colors light / dark mode variant by calling:
```swift
let lightColor = UIColor(.keyPressed).light
let darkColor = UIColor(.keyPressed).dark
```

I updated the Changelog as well after merging in the current master branch into my branch. However, I am not sure if I placed it under the correct version. I wasn't sure if 2.0.0 is already released now.

As a second step all color variables throughout the project could also be replaced with calls to the enum's `color` property or calls to the new `UIColor` init. However, this would lead to lots of changes which I did not want to do now.

Feel free to change anything you want :) If you want me to change anything, please comment as well.

### CODE REVIEW

**Comment by andrewtavis** [comment]
Looking at all this now, @misteu! Sorry for the wait 😊

File: Keyboards/KeyboardsBase/ColorVariables.swift
Comment (andrewtavis) [inline Keyboards/KeyboardsBase/ColorVariables.swift:30]: This is beautiful :) :) I'm already looking forward to all these changes just given the deletions!

File: Scribe/AppStyling.swift
Comment (andrewtavis) [inline Scribe/AppStyling.swift:14]: Nice :) I'm gonna do #227 after this to give these changes a try 😊

**Comment by andrewtavis** [comment]
Wow and it all looks so organized in Xcode as well!

**Comment by andrewtavis** [comment]
@misteu, just wanted to check what the next steps on this are. When I load the keyboard, all I see is a red screen where it used to be. Let me know what all I need to do, and then am happy to approve, merge, and make the necessary edits 😊

**Comment by misteu** [comment]
> @misteu, just wanted to check what the next steps on this are. When I load the keyboard, all I see is a red screen where it used to be. Let me know what all I need to do, and then am happy to approve, merge, and make the necessary edits 😊

oh sorry that should bot be the case! I had this state a little earlier, I might have forgotten to push something.

Red basically means: Some color could not be loaded. I think i also added a print statement to the debugger whenever this happens. By that it should be easier to see which or if none of the colors could be loaded.

Unfortunately, I do not have time to look into it this weekend. I will have a look next week (Hopefully, as soon as Monday 🙂 )

What you can do already is to check if the assets file (which now includes images and colors) is added to the correct targets. I think it needs to be added to all the keyboard extensions (might be gone when I solved a merge conflict or something)

Anyway, let me know if you still have problems with it :) I'll have a look next week as I said.

Have a nice weekend!

**Comment by andrewtavis** [comment]
@misteu, the "red alert" for the color makes a lot of sense, and will doubtless come in handy later on :) I'm doing some Fall exploration this weekend, so let's just check in on this when you've done the correction next week 😊

I'll make a note for myself below though:
- [ ] The assets file has been added to the correct targets

Have a wonderful weekend yourself!

File: CHANGELOG.md
Comment (wkyoshida) [inline CHANGELOG.md:61]: > I updated the Changelog as well after merging in the current master branch into my branch. However, I am not sure if I placed it under the correct version. I wasn't sure if 2.0.0 is already released now.

Hey @misteu! Wanted to add a quick note regarding the above :arrow_up: so that what you brought up doesn't get overlooked :+1:. Looks like [2.0.0](https://github.com/scribe-org/Scribe-iOS/releases/tag/2.0.0) did already get released, only couple days ago. However, it does look like @andrewtavis did just recently add now the next version in the `CHANGELOG.md` as well over [here](https://github.com/scribe-org/Scribe-iOS/blob/main/CHANGELOG.md#scribe-ios-210). My guess is that what you added would go in there instead.

@andrewtavis - just a thought that I had, but could it be good to note which version is currently in development? e.g. the title of the current subsection in the `CHANGELOG.md` could perhaps be 

### [Unreleased] Scribe-iOS 2.1.0 

or

### [In Progress] Scribe-iOS 2.1.0

or even just simply

### [Unreleased]

or something else of the like. Just an idea, since I think it could help clarify, in case other contributors wonder the same thing. Also, since there can often be cleanup of the `CHANGELOG.md` right before a release is made regardless too, I figured it could be easy enough to do the rename to the actual version title then. 

Have a great weekend all!

Comment (andrewtavis) [reply CHANGELOG.md:61]: I totally agree, @wkyoshida :) I think that simply marking it as `[Unreleased]` if we're unsure if it's going to be a minor or a patch release, and as soon as we figure out what kind it is we can add in the version. The upcoming one adds at least one feature, so we can mark it as `[Unreleased] Scribe-iOS 2.1.0` for now 😊

Thanks for your further contributions, and hope both of you are having a nice weekend so far!

**Comment by andrewtavis** [comment]
@misteu, [1393f65](https://github.com/scribe-org/Scribe-iOS/pull/228/commits/1393f65500d77520df5e4edfb7614803922023af) was to fix the conflicts as there are `preposition.json` files for each keyboard now regardless of if the language has preposition-case relationships (allows for us to remove all the optionals from data references) :)

File: CHANGELOG.md
Comment (andrewtavis) [reply CHANGELOG.md]: [521976d](https://github.com/scribe-org/Scribe-iOS/pull/228/commits/521976dd54db99cf62ee80686365292a200c50a8) makes the changes that we discussed here 😊

**Comment by misteu** [comment]
> @misteu, the "red alert" for the color makes a lot of sense, and will doubtless come in handy later on :) I'm doing some Fall exploration this weekend, so let's just check in on this when you've done the correction next week 😊
> 
> I'll make a note for myself below though:
> 
>     * [ ]  The assets file has been added to the correct targets
> 
> 
> Have a wonderful weekend yourself!

Hey @andrewtavis !
I just had some time for a quick look. Looks like I forgot to add the assets to all the targets in my pushed changes:
image

I'll push a fix for that in a minute

**Comment by andrewtavis** [comment]
Yes @misteu, I saw that as well as was trying to fix it :) I'm still getting used to committing to other branches, so I screwed up on my end 😅 Happy to merge when that change comes through!

**Review by andrewtavis — APPROVED** [review]
All looks good to me now, @misteu! Thanks for your help with this! Looking forward to working more with the new color system going forward 😊

**Comment by misteu** [comment]
OK :) I just pulled the latest changes there and added the fix (checking all the targets).

Only strange thing is, that I cannot have Xcodes debugger attached together with the keyboard. It always stops (not sure if it really crashes) in some breakpoints.

**Comment by andrewtavis** [comment]
Potentially what's going on there is that you have `Scribe` as the app you're running? We had tons of problems with running the debugger until I figured out that we need to select the specific keyboard we're debugging as the target. Then it works 😊

**Pull request merged**
